sdk/monitor/monitor-ingestion/src/utils/concurrentPoolHelper.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+
+export async function concurrentRun<T>(
+  maxConcurrency: number,
+  inputData: Array<T>,
+  callback: (args: T) => Promise<void>
+): Promise<void> {
+    const dataQueue = [...inputData].reverse();
+    const promises: Array<Promise<void>> = [];
+    
+    function removePromise(p: Promise<void>) {
+      promises.splice(promises.indexOf(p),1);
+    }
+    while(dataQueue.length) {
+      while(promises.length < maxConcurrency) {
```

Comment:
To fix the issue Deya mentioned
```suggestion
      while(dataQueue.length && promises.length < maxConcurrency) {
```

Reply:
done